lib/matplotlib/patches.py

```diff
@@ -1113,6 +1118,11 @@ def __init__(self, x, y, dx, dy, width=0.001, length_includes_head=False,
             head_length = 1.5 * head_width
 
         distance = np.sqrt(dx ** 2 + dy ** 2)
+
+        # Account for divide by zero
+        if distance == 0:
```

Comment:
Yes, but it would be clearer and more direct to replace those last two lines with `cx, sx = 0, 1`.   All of this is not fundamentally a matter of handling a possible "divide by zero", it is handling a situation in which the length of a vector is zero so there is no information about orientation; the orientation is arbitrary.